src/gap.c

```diff
@@ -876,6 +876,31 @@ Obj FuncRUNTIMES( Obj     self)
 }
 
 
+
+/****************************************************************************
+**
+*F  FuncGetTimeOfDay( <self> )
+**
+**  'FuncGetTimeOfDay' implements the function 'GetTimeOfDay'.
+**
+**  'GetTimeOfDay' returns the current time since 1.1.1970 in seconds and
+**  microseconds as per the POSIX specification.
+**  The accuracy of this is system dependent.
+** 
+**  Also note that gettimeofday has been marked obsolete in the POSIX
+**  standard.
+*/
+Obj FuncGetTimeOfDay(Obj self)
+{
+  Obj tmp;
+  struct timeval t;
+  gettimeofday(&t, NULL);
+  tmp = NEW_PREC(0);
+  AssPRec(tmp, RNamName("tv_sec"), ObjInt_Int(t.tv_sec));
+  AssPRec(tmp, RNamName("tv_usec"), ObjInt_Int(t.tv_usec));
```

Comment:
Why not go the full way and also extract and return the timezone data here? Granted, we don't need it right now, but if somebody needs it in the future, we don't have to add a second function (or modify this one retroactively).

One reason *not* to do this might be portability concerns: Emulating this function on e.g. Windows might be annoying.

But if that is the concern, why not implementing `MicrosecondsSinceEpoch` in C, thus hiding this POSIX specific call?